### PR TITLE
docs: New protocol specification

### PR DESCRIPTION
**Pull request by kelemeno**
## What ❔

New docs, focusing on protocol specs.  

Note: don't merge yet, waiting for review.

## Why ❔

We want to have good docs so people can understand the zkVM and its benefits. 

## Checklist

- [ X] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ X] Tests for the changes have been added / updated.
- [ X] Documentation comments have been added / updated.
- [ X] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `cargo spellcheck --cfg=./spellcheck/era.cfg --code 1`.

### CODE REVIEW

**Review by shahar4 — DISMISSED** [review]

**Comment by mm-zk** [comment]
BTW - any reason why you removed docs/advanced/blocks_and_batches.md instead of renaming?

File: docs/guides/advanced/0_alternative_vm_intro.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+For efficiency reasons it the zkEVM is similar to the EVM. This makes executing smart programs inside of it easy. It
+also has special features that are not in the EVM but are needed for the rollup's STF, storage, gas metering,
+precompiles and other things. Some of these features are implemented as system contracts while others are built into the
+VM. System Contracts are contracts with special permissions, deployed at predefined addresses. Finally, we have the
+bootloader, which is also a contract, although it is not deployed at any address. This is the STF that is ultimately
+executed by the zkEVM, and executes the transaction against the state.
+
+<!-- KL todo *Add different abstraction levels diagram here:*-->
+
+Full specification of the zkEVM is beyond the scope of this document. However, this section will give you most of the
```
Comment (kelemeno) [inline docs/guides/advanced/0_alternative_vm_intro.md:26]: @shahar4 this doc until ## registers and memory (from then on its copied)

File: docs/specs/data_availability/validium_zk_porter.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Validium and zkPorter
```
Comment (kelemeno) [inline docs/specs/data_availability/validium_zk_porter.md:1]: This doc @shahar4

File: docs/specs/l1_l2_communication/overview_deposits_withdrawals.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Overview - Deposits and Withdrawals
```
Comment (kelemeno) [inline docs/specs/l1_l2_communication/overview_deposits_withdrawals.md:1]: @shahar4

File: docs/specs/overview.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Overview
```
Comment (kelemeno) [inline docs/specs/overview.md:1]: @shahar4

File: docs/specs/the_hyperchain/hyperbridges.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Hyperbridges
```
Comment (kelemeno) [inline docs/specs/the_hyperchain/hyperbridges.md:1]: @shahar4

File: docs/specs/the_hyperchain/overview.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Overview
```
Comment (kelemeno) [inline docs/specs/the_hyperchain/overview.md:1]: @shahar4

**Review by shahar4 — APPROVED** [review]

**Pull request merged**